helm.tf

```diff
@@ -99,191 +82,830 @@ resource "helm_release" "metrics_server" {
+  namespace         = "monitoring"
+  create_namespace  = true
+  repository        = "https://prometheus-community.github.io/helm-charts"
+  chart             = "kube-prometheus-stack"
+  version           = "35.0.3"
+  dependency_update = true
+  timeout           = 600
+
+  set {
+    name = "grafana.enabled"
+    value = false
+  }
+
+  set {
+    name  = "prometheus.prometheusSpec.replicas"
+    value = var.prometheus_replicas
+  }  
+
+  dynamic "set" {
+    for_each = var.prometheus_requests_cpu != null ? ["do it"] : []
+    content {
+      name  = "prometheus.prometheusSpec.resources.requests.cpu"
+      value = var.prometheus_requests_cpu
+    }
+  }
+
+  dynamic "set" {
+    for_each = var.prometheus_requests_ram != null ? ["do it"] : []
+    content {
+      name  = "prometheus.prometheusSpec.resources.requests.memory"
+      value = var.prometheus_requests_ram
+    }
+  }
+
+  dynamic "set" {
+    for_each = var.prometheus_limits_cpu != null ? ["do it"] : []
+    content {
+      name  = "prometheus.prometheusSpec.resources.limits.cpu"
+      value = var.prometheus_limits_cpu
+    }
+  }
+
+  dynamic "set" {
+    for_each = var.prometheus_limits_ram != null ? ["do it"] : []
+    content {
+      name  = "prometheus.prometheusSpec.resources.limits.memory"
+      value = var.prometheus_limits_ram
+    }
+  }
+
+  set {
+    name  = "prometheus.prometheusSpec.retention"
+    value = var.prometheus_metrics_retention
+  }  
+
+  set {
+    name  = "prometheus.ingress.enabled"
+    value = var.prometheus_ingress_enabled
+  }  
+
+  set {
+    name  = "prometheus.ingress.hosts[0]"
+    value = var.prometheus_ingress_host
+  }  
+
+  set {
+    name  = "prometheus.ingress.paths[0]"
+    value = var.prometheus_ingress_path
+  }  
+
+  set {
+    name  = "prometheus.ingress.pathType"
+    value = var.prometheus_ingress_path_type
+  }  
+
+  set {
+    name  = "prometheus.ingress.ingressClassName"
+    value = var.prometheus_ingress_class_name
+  }  
+
+  set {
+    name  = "prometheus.prometheusSpec.storageSpec.volumeClaimTemplate.spec.storageClassName"
+    value = var.prometheus_storage_class_name
+  }  
+
+
+  set {
+    name  = "prometheus.prometheusSpec.storageSpec.volumeClaimTemplate.spec.resources.requests.storage"
+    value = var.prometheus_storage_size
+  }  
+
+
+  set {
+    name  = "prometheus.prometheusSpec.storageSpec.volumeClaimTemplate.spec.accessModes[0]"
+    value = "ReadWriteOnce"
+  }  
+
+
+  depends_on = [time_sleep.wait_20_seconds]
+
+}
+
+# ================== loki-distributed ================== #
+
+resource "helm_release" "loki_distributed" {
+  count             = var.helm_loki_enabled ? 1 : 0
+  name              = "loki-distributed"
   namespace         = "monitoring"
   create_namespace  = true
   repository        = "https://grafana.github.io/helm-charts"
-  chart             = "loki-stack"
-  version           = "2.4.1"
+  chart             = "loki-distributed"
+  version           = "0.48.3"
```

Comment:
We should allow chart version selection (module input)

Reply:
I'm marking this as resolved because we agreed to don't do it